src/Components/UserSharedComponents/BookingForm/BookingForm.tsx

```diff
@@ -51,8 +68,34 @@ function BookingForm({ room }: BookingFormProps) {
     };
 
     console.log(bookingDetails);
-
-    /// check login and call api
+    const isLoggedIn = Boolean(localStorage.getItem("token"));
```

Comment:
This will get evaluated whenever the components re-render which is not good as the localStorage access is an expensive operation. We can make it  
```tsx
const [isLoggedIn] = React.useState(()=> localStorage.getItem("token))
```

The initializer function will only run in the first render instead of getting evaluated in every render.
We also will need to make multiple checks as the localStorage might return null if there is not value stored, so We should address this as well.